### ISSUE TITLE
Unsound method casts mutable reference to pointer

### ISSUE THREAD

**Issue by fayalalebrun** [ViliamVadocz/nvidia-video-codec-sdk#18]
The method in question is part of `NV_ENC_INITIALIZE_PARAMS` and can be found here: https://docs.rs/nvidia-video-codec-sdk/latest/src/nvidia_video_codec_sdk/safe/builders.rs.html#54-57

The problem is that this method takes a mutable reference to an `NV_ENC_CONFIG`. However, this is then stored to a pointer. Therefore, the `NV_ENC_CONFIG` member may disappear at some point while not violating any lifetime constraints.

This effectively means the `NV_ENC_CONFIG` may contain random data, causing strange, difficult to debug errors. Or no errors at all in some cases.

**Comment by ViliamVadocz** [comment]
Good point. To do this correctly I (or any other contributor) should make a safe wrapper around `NV_ENC_INITIALIZE_PARAMS` that stores the actual reference, and any method that currently uses the params should take this wrapper instead, and convert to the raw pointer version.

Something similar was done with [`EncodePictureParams`](https://docs.rs/nvidia-video-codec-sdk/latest/nvidia_video_codec_sdk/safe/struct.EncodePictureParams.html), although there I did not have any references, but at least it shows the idea of how to safely wrap the params.

You, are anyone else, is free to work on this and I will happily review the pull request. I am a bit busy with university at the moment so I cannot dedicate time to do it myself yet.

**Comment by ViliamVadocz** [comment]
Fixed by #19.
